### DISCUSSION TITLE
Use of BUZZER with the ESP32-WebCam version (standard and customized)

### DISCUSSION THREAD

**Discussion by Bebuccio** [arendst/Tasmota#18576] [Q&A] [answered]
I want to use a buzzer device connected to a ESP32-webcam board.
Using the standard "http://ota.tasmota.com/tasmota32/release/tasmota32-webcam.bin" version, there isn't the buzzer between the devices allowed to set to GPIO pin.
Since I have to use script indeed of rules, I have made a customized version of webcam fw (#undef USE_RULES #define USE_SCRIPT).
To include the buzzer driver, I have inserted also the "#define USE_BUZZER" into the user_config_override.h files but, after loading this fw, I have the scrpt section but I don't found the buzzer device.
There is an incompatibility between webcam and buzzer device into the ESP32?

**Comment by sfromis** [comment]
There is an "incompatibility" between running the very processor intensive webcam feature, and assuming that "any other" feature can work well. Hence only very few features are included along with webcam.

More generally, when you use *any* of the "premixed" tasmota32-xxxxxx configurations, most of your feature amendments in `user_config_override.h` will be ignored. You should have read:
https://tasmota.github.io/docs/Compile-your-build/#customize-your-build

**Reply by Bebuccio** [reply]
Thanks sfromis, but I have not undestand.
There is a way to use webcam and buzzer together?

**Reply by sfromis** [reply]
Nothing stops you from digging into the build process to change whatever, outside of the default easy build setup. Do **read** the page I already linked.

And it may impede operation to combine features. There are reasons why the webcam build deliberately was cut strongly down.

**Reply by Bebuccio** [reply]
Sorry sformis.
I have read carefully the document "https://tasmota.github.io/docs/Compile-your-build/#customize-your-build" but I have not found anything about what incompatibility are present in a premixed configuration.
So also the standard version don't have the buzzer device.
I have found into the DB of discussion this thread: https://github.com/arendst/Tasmota/discussions/15892
Here you have answered that in the webcam versione of tasmota 32 is possible to use the buzzer.
But I don't find it into the standard version http://ota.tasmota.com/tasmota32/release/tasmota32-webcam.bin

**Reply by barbudor** [reply]
![Screenshot_2023-05-04-11-21-16-686_com android chrome-edit](https://user-images.githubusercontent.com/2996042/236163528-0e9cce77-7625-448e-8535-0e04e1c34be8.jpg)

This what sfromis is referring to

**Reply by sfromis** [reply]
> Here you have answered that in the webcam versione of tasmota 32 is possible to use the buzzer.

No. I only pointed out that if you read the full page, you'll find information on how to modify builds, to a deeper degree than the "entry level" simple modifications via `user_config_override.h`. And that if you do that, you're on your own if it does not work well with feature combinations deliberately left out of the standard builds.

**Reply by Bebuccio** [reply]
I have used the user_config_override.h  with these defines
#ifndef USE_WEBCAM
  #define USE_WEBCAM
#endif

#ifndef CAMERA_MODEL_AI_THINKER
  #define CAMERA_MODEL_AI_THINKER
#endif_

Then I have compiled using the command
**platformio run -e tasmota32**

Loaded the fw on the board I see the webcam but the PSRAM is disabled.
The console shows:
00:00:00.002 HDW: ESP32-D0WDQ6 (**PSRAM disabled**)
00:00:00.003 HDW: PSRAM is disabled, requires specific compilation on this hardware (see doc)

**How I have to compile the fw to use PSRAM?
Which platform_xxxx_.ini I have to modify?
And in which manner?**

Thanks

**Reply by Jason2866** [reply, marked as answer]
PSRAM is disabled when the PSRAM fix is needed. You see a message in the log.
In this case the board `esp32-fix` needs to be choosen.
You can override here
https://github.com/arendst/Tasmota/blob/development/platformio_override_sample.ini#L98
